Client/Frontend/Settings/LocationViewPositionPickerCell.swift

```diff
@@ -0,0 +1,127 @@
+
+/// Displays a picker that allows the user to pick the position of the location view (URL bar)
+class LocationViewPositionPickerCell: UITableViewCell, Cell {
+  private let stackView = UIStackView().then {
+    $0.spacing = 12
+    $0.distribution = .fillEqually
+  }
+  
+  private let topBarButton = BarButton(
+    image: UIImage(braveSystemNamed: "brave.platter.filled.top.and.arrow.up.phone")!,
+    text: "Top Bar"
+  )
+  
+  private let bottomBarButton = BarButton(
+    image: UIImage(braveSystemNamed: "brave.platter.filled.bottom.and.arrow.down.phone")!,
+    text: "Bottom Bar"
+  )
+  
+  override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+    super.init(style: style, reuseIdentifier: reuseIdentifier)
+    
+    contentView.addSubview(stackView)
+    stackView.addArrangedSubview(topBarButton)
+    stackView.addArrangedSubview(bottomBarButton)
+    
+    let isUsingBottomBar = Preferences.General.isUsingBottomBar.value
+    topBarButton.isSelected = !isUsingBottomBar
+    bottomBarButton.isSelected = isUsingBottomBar
+    
+    stackView.snp.makeConstraints {
+      $0.leading.trailing.equalToSuperview().inset(24)
+      $0.top.equalToSuperview().inset(12)
+      $0.bottom.equalToSuperview().inset(16)
+    }
+    
+    topBarButton.addTarget(self, action: #selector(tappedTopBarButton), for: .touchUpInside)
+    bottomBarButton.addTarget(self, action: #selector(tappedBottomBarButton), for: .touchUpInside)
+  }
+  
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError()
+  }
+  
+  @objc private func tappedTopBarButton() {
+    topBarButton.isSelected = true
+    bottomBarButton.isSelected = false
+    Preferences.General.isUsingBottomBar.value = false
+  }
+  
+  @objc private func tappedBottomBarButton() {
+    topBarButton.isSelected = false
+    bottomBarButton.isSelected = true
+    Preferences.General.isUsingBottomBar.value = true
+  }
+  
+  class BarButton: UIControl {
+    var image: UIImage
+    var text: String
+    
+    private let checkmarkView = UIImageView().then {
+      $0.preferredSymbolConfiguration = .init(font: .preferredFont(forTextStyle: .title3))
+    }
+    
+    init(image: UIImage, text: String) {
+      self.image = image
+      self.text = text
+      
+      super.init(frame: .zero)
+      
+      let stackView = UIStackView()
+      stackView.isUserInteractionEnabled = false
+      stackView.axis = .vertical
+      stackView.alignment = .center
+      stackView.spacing = 2
+      
+      let imageView = UIImageView(image: image)
+      imageView.preferredSymbolConfiguration = .init(font: .systemFont(ofSize: 56))
+      let label = UILabel()
+      label.text = text
+      label.font = .preferredFont(forTextStyle: .callout)
+      
+      addSubview(stackView)
+      stackView.addArrangedSubview(imageView)
+      stackView.addArrangedSubview(label)
+      stackView.addArrangedSubview(checkmarkView)
+      stackView.setCustomSpacing(8, after: label)
```

Comment:
Maybe use the `addStackViewItems(.view()/.spacing)` helper method you added back in the day

Reply:
Done!